pkg/appfile/parser.go

```diff
@@ -171,9 +183,114 @@ func (p *Parser) parseTrait(name string, properties map[string]interface{}) (*Tr
 	}
 
 	return &Trait{
-		Name:     name,
-		Params:   properties,
-		Template: templ,
-		Health:   health,
+		Name:               name,
+		CapabilityCategory: capabilityCategory,
+		Params:             properties,
+		Template:           templ,
+		Health:             health,
 	}, nil
 }
+
+// GenerateApplicationConfiguration converts an appFile to applicationConfig & Components
```

Comment:
Is this part new or just moving from the application controller?

Reply:
Moved from application controller.